src/libs/actions/TestTool.ts

```diff
@@ -14,7 +15,13 @@ Onyx.connect({
  * Throttle the toggle to make the modal stay open if you accidentally tap an extra time, which is easy to do.
  */
 function toggleTestToolsModal() {
-    const toggle = () => Onyx.set(ONYXKEYS.IS_TEST_TOOLS_MODAL_OPEN, !isTestToolsModalOpen);
+    const toggle = () => {
+        if (!isTestToolsModalOpen) {
+            Modal.close(() => Onyx.set(ONYXKEYS.IS_TEST_TOOLS_MODAL_OPEN, !isTestToolsModalOpen));
+            return;
+        }
+        Onyx.set(ONYXKEYS.IS_TEST_TOOLS_MODAL_OPEN, !isTestToolsModalOpen);
+    };
```

Comment:
The issue is that, when pressing the shortcut, we use `Modal.close` which will close any active modal first. When we press the shortcut for the 2nd time, the debug modal will be closed first and after it's closed we toggle the visibility. But in the test tool modal component, we also toggle the visibility when hidden.
https://github.com/Expensify/App/blob/b05ae2f4146af68391f57bbf8fb9c73459de6c73/src/components/TestToolsModal.tsx#L33